src/utils/mixins.js

```diff
@@ -183,60 +186,64 @@ const fileUploader = {
         } else {
           const progress = (event) => {
-            Object.assign(file.progress, event);
+            this.$set(file, 'progress', event);
           };
           file.status = 'uploading';
           file.progress.indeterminate = true;
-          try {
-            if (file.upload) {
-              // eslint-disable-next-line no-await-in-loop
-              file.result = await file.upload.resume();
-            } else {
-              // eslint-disable-next-line new-cap
-              file.upload = new uploadCls(file.file, {
-                $rest: this.girderRest,
-                parent: dest,
-                progress,
-              });
-              // eslint-disable-next-line no-await-in-loop
-              await file.upload.beforeUpload();
-              // eslint-disable-next-line no-await-in-loop
-              file.result = await file.upload.start();
-            }
-            // eslint-disable-next-line no-await-in-loop
-            await file.upload.afterUpload();
-            delete file.upload;
-            results.push(file.result);
-            file.status = 'done';
-            file.progress.current = file.file.size;
-          } catch (error) {
-            // eslint-disable-next-line no-await-in-loop
-            await file.upload.onError(error);
-
-            if (error.response) {
-              this.errorMessage = error.response.data.message || 'An error occurred during upload.';
-            } else {
-              this.errorMessage = 'Connection failed.';
-            }
-            file.status = 'error';
-            this.uploading = false;
-            this.$emit('error', { error, file });
-            return;
+          if (file.upload) {
+            promiseChain = promiseChain.then(() => file.upload.resume());
+          } else {
+            // eslint-disable-next-line new-cap
+            file.upload = new uploadCls(file.file, {
+              $rest: this.girderRest,
+              parent: dest,
+              progress,
+            });
+            promiseChain = promiseChain
+              .then(() => file.upload.beforeUpload())
+              .then(() => file.upload.start());
           }
+          promiseChain = promiseChain
+            .then(async (result) => {
+              await file.upload.afterUpload();
+              delete file.upload;
+              file.status = 'done';
+              file.progress.current = file.file.size;
+              return result;
+            })
+            .catch(async (error) => {
+              await file.upload.onError(error);
+              if (error.response) {
+                this.errorMessage = error.response.data.message || 'An error occurred during upload.';
+              } else {
+                this.errorMessage = 'Connection failed.';
+              }
+              file.status = 'error';
+              this.uploading = false;
+              this.$emit('error', { error, file });
+              throw error;
```

Comment:
This is behavior is different from what was here before.  Now, unless the caller of `start()` handles errors, this will be uncaught.   I tend to prefer that unhandled errors don't get swallowed by this library.

It will also be uncaught in Upload.vue.  I don't really see that as a problem, but we can catch it there if necessary.

Reply:
I like the idea of throwing here, however, shouldn't we handle it in Upload.vue since that's where `start` is called? Otherwise wouldn't the only way to catch an exception during upload be to manually call `startUpload` from the outside?

Reply:
> shouldn't we handle it in Upload.vue since that's where start is called

Sure, an empty handler can be added to Upload.vue.

Reply:
Sorry, I may be misunderstanding something. Prior to this change, my understanding was that the Upload.vue *did* handle the errors, by presenting the error message, and allowing "Resume" to be clicked. Was that not the case?

Reply:
No.  prior to this change, `start()` swallowed the error and expected the mixin consumer to present information about the error using `this.errorMessage`.  `errorMessage` is useful, but I also think `start()` should throw the exception too in case some external mixin consumer has other handling they want to do.

Reply:
I should say `start()` swallowed _some_ of the errors.  preUpload and postUpload were unhandled, but beforeUpload, afterUpload, and all axios errors were handled.